### ISSUE TITLE
Assistive Access broken from Alacritty

### ISSUE THREAD

**Issue by matan129** [alacritty/alacritty#7334]
### System

OS: macOS Ventura 13.6.1, Apple Silicon (M2)
Version: `alacritty 0.12.3 (5efb069)`

### Issue

Hi folks, I'm facing a weird issue.

I have an AppleScript script which automates some GUI app. For that, it needs Assistive Access permissions (aka Accessibility). 
I've granted Alacritty, `osascript` this permission:
![image](https://github.com/alacritty/alacritty/assets/1483099/94e53121-bbef-4b13-8153-760401c6a29b)

From any other terminal _except_ Alacritty, it works flawlessly.
On Alacritty, I'm getting 
```
$ osascript ~/path/to/script.scpt
~/path/to/script.scpt: execution error: System Events got an error: osascript is not allowed assistive access. (-1719)
```
Again, under any other terminal (built-in, iTerm, Rio, etc.), this doesn't happen. 
If I create a tmux session from any non-Alacritty terminal and then attach to it via Alacritty, the problem disappears.

I've tried revoking the Accessability permission from Alacritty and re-grating it; uninstalling and re-installing, etc., but to no avail.
Not sure how to debug this further.
I thought it might be related to macOS app signatures, but AFIAK Rio is not signed either so I don't see how it's different.

I'd appreciate any help on this matter (including making PRs as needed).

Thanks a bunch!

**Comment by nixpulvis** [comment]
Could it be that you have multiple installs of Alacritty?

Is it any `osascript` invocation, or just a specific script?

**Comment by matan129** [comment]
Nope, single installation. I made sure that the Alacritty binary I gave permissions to is the actual one I'm running (I checked its path via Activity Monitor).

The issue affect all scripts that actually try and use Assistive stuff:
```AppleScript
tell application "System Events"
	tell process "JetBrains Toolbox" (* Choose any other process that has a menu bar icon *)
		click menu bar item 1 of menu bar 2
	end tell
end tell
```

Other scripts work fine, like 
```AppleScript
tell application "System Events"
	tell process "JetBrains Toolbox" (* Choose any other process that has a menu bar icon *)
                (* Not clicking on anything, so it works *)
		log "hi"
	end tell
end tell
```

**Comment by nixpulvis** [comment]
I'm running macOS 14 to test this and it seems to work for me. It may be worth noting that I don't need to grant `osascript` permissions, I simply give Alacritty the permission.

**Comment by nixpulvis** [comment]
Only thing I can really suggest at this point is to uninstall all copies of Alacritty you have and reinstall one and see if it works.

Sorry, I don't really know much about this feature of macOS myself.

**Comment by matan129** [comment]
Already tried that, but I'll try again. Thanks... I'll keep this open just in case somebody stumbles upon this with an idea.

**Comment by matan129** [comment]
Alright, I managed to go around it with 

1. Resetting the entire a11y permissions with `sudo tccutil reset Accessibility`
2. Restarting the machine
3. Re-granting Alacritty (& all of the other apps) Accessibility permissions via System Settings.

Now everything works again :tada:

At this point, I'm pretty sure that it's some macOS shenanigans and not Alacritty.
Thanks @nixpulvis!

**Comment by AlmastSargsyan0615** [comment]
"System Events got an error: Application is not allowed assistive access. (-1719)"

I also faced this error.

This is my solution:
https://github.com/AlmastSargsyan0615/Safari-Automation-using-Applecript/blob/main/video.mp4

To resolve this error, I tried to search solution from online but their solution was not working.

https://stackoverflow.com/questions/36209964/applescript-issue-not-allowed-assistive-access-1719

Especially, they said about the such as above solution but hmm.

To resolve this error, I made shell script automator and this was worked.

**Comment by landonb** [comment]
Thanks, @matan129 ! Your solution worked!

Warning to others: After `sudo tccutil reset Accessibility`, my machine become unresponsive to mouse and keyboard events, and I had to cold reset.

FYI, I was getting the specific error:

```
$ osascript ~/path/to/script.osa
/Users/user/path/to/script.osa:2389:2670: execution error:
  System Events got an error: osascript is not allowed assistive access. (-25211)
```

I first disabled and enabled Alacritty *Accessibility* access, then rebooted, but that didn't work.

I did *not* try to uninstall and reinstall Alacritty. Also, I could run my script fine through iTerm.

Thanks again and I'm glad I found this issue!
